File: protocol/src/main/resources/typescript/IByteBuffer.ts
interface IByteBuffer {
    adjustPadding(predictionLength: number, beforeWriteIndex: number): void
    compatibleRead(beforeReadIndex: number, length: number): boolean
    setWriteOffset(writeOffset: number): void
    getWriteOffset(): number
    setReadOffset(readOffset: number): void
    getReadOffset(): number
    getCapacity(): number
    ensureCapacity(minCapacity: number): void
    isReadable(): boolean
    writeBytes(byteArray: ArrayBuffer): void
    toBytes(): ArrayBuffer
    writeBoolean(value: boolean): void
    readBoolean(): boolean
    writeByte(value: number): void
    readByte(): number
    writeShort(value: number): void
    readShort(): number
    writeRawInt(value: number): void
    readRawInt(): number
    writeInt(value: number): void
    writeIntCount(value: number): number
    readInt(): number
    writeLong(value: number): void
    readLong(): number
    writeFloat(value: number): void
    readFloat(): number
    writeDouble(value: number): void
    readDouble(): number
    writeString(value: string): void
    readString(): string
    writePacket(packet: any, protocolId: number): void
    readPacket(protocolId: number): any
    writeBooleanArray(array: Array<boolean> | null): void
    readBooleanArray(): boolean[]
    writeByteArray(array: Array<number> | null): void
    readByteArray(): number[]
    writeShortArray(array: Array<number> | null): void
    readShortArray(): number[]
    writeIntArray(array: Array<number> | null): void
    readIntArray(): number[]
    writeLongArray(array: Array<number> | null): void
    readLongArray(): number[]
    writeFloatArray(array: Array<number> | null): void
    readFloatArray(): number[]
    writeDoubleArray(array: Array<number> | null): void
    readDoubleArray(): number[]
    writeStringArray(array: Array<string> | null): void
    readStringArray(): string[]
    writePacketArray(array: Array<any> | null, protocolId: number): void
    readPacketArray(protocolId: number): any
    // ---------------------------------------------list-------------------------------------------
    writeBooleanList(list: Array<boolean> | null): void
    readBooleanList(): boolean[]
    writeByteList(list: Array<number> | null): void
    readByteList(): number[]
    writeShortList(list: Array<number> | null): void
    readShortList(): number[]
    writeIntList(list: Array<number> | null): void
    readIntList(): number[]
    writeLongList(list: Array<number> | null): void
    readLongList(): number[]
    writeFloatList(list: Array<number> | null): void
    readFloatList(): number[]
    writeDoubleList(list: Array<number> | null): void
    readDoubleList(): number[]
    writeStringList(list: Array<string> | null): void
    readStringList(): string[]
    writePacketList(list: Array<any> | null, protocolId: number): void
    readPacketList(protocolId: number): any[]
    // ---------------------------------------------set-------------------------------------------
    writeBooleanSet(set: Set<boolean> | null): void
    readBooleanSet(): Set<boolean>
    writeByteSet(set: Set<number> | null): void
    readByteSet(): Set<number>
    writeShortSet(set: Set<number> | null): void
    readShortSet(): Set<number>
    writeIntSet(set: Set<number> | null): void
    readIntSet(): Set<number>
    writeLongSet(set: Set<number> | null): void
    readLongSet(): Set<number>
    writeFloatSet(set: Set<number> | null): void
    readFloatSet(): Set<number>
    writeDoubleSet(set: Set<number> | null): void
    readDoubleSet(): Set<number>
    writeStringSet(set: Set<string> | null): void
    readStringSet(): Set<string>
    writePacketSet(set: Set<any> | null, protocolId: number): void
    readPacketSet(protocolId: number): Set<any>
    // ---------------------------------------------map-------------------------------------------
    writeIntIntMap(map: Map<number, number> | null): void
    readIntIntMap(): Map<number, number>
    writeIntLongMap(map: Map<number, number> | null): void
    readIntLongMap(): Map<number, number>
    writeIntStringMap(map: Map<number, string> | null): void
    readIntStringMap(): Map<number, string>
    writeIntPacketMap(map: Map<number, any> | null, protocolId: number): void
    readIntPacketMap(protocolId: number): Map<number, any>
    writeLongIntMap(map: Map<number, number> | null): void
    readLongIntMap(): Map<number, number>
    writeLongLongMap(map: Map<number, number> | null): void
    readLongLongMap(): Map<number, number>
    writeLongStringMap(map: Map<number, string> | null): void
    readLongStringMap(): Map<number, string>
    writeLongPacketMap(map: Map<number, any> | null, protocolId: number): any
    readLongPacketMap(protocolId: number): Map<number, any>
    writeStringIntMap(map: Map<string, number> | null): void
    readStringIntMap(): Map<string, number>
    writeStringLongMap(map: Map<string, number> | null): void
    readStringLongMap(): Map<string, number>
    writeStringStringMap(map: Map<string, string> | null): void
    readStringStringMap(): Map<string, string>
    writeStringPacketMap(map: Map<string, any> | null, protocolId: number): void
    readStringPacketMap(protocolId: number): Map<string, any>
}

export default IByteBuffer;

File: protocol/src/main/resources/dart/IByteBuffer.dart
import 'dart:typed_data';

abstract class IByteBuffer {
  void adjustPadding(int predictionLength, int beforeWriteIndex);

  bool compatibleRead(int beforeReadIndex, int length);

  Int8List getBuffer();

  int getWriteOffset();

  void setWriteOffset(int writeIndex);

  int getReadOffset();

  void setReadOffset(int readIndex);

  bool isReadable();

  int getCapacity();

  void ensureCapacity(int capacity);

  void writeBytes(Int8List bytes);

  Int8List readBytes(int length);

  void writeBool(bool value);

  bool readBool();

  void writeByte(int value);

  int readByte();

  void writeShort(int value);

  int readShort();

  int writeIntCount(int value);

  void writeRawInt(int value);

  int readRawInt();

  void writeInt(int value);

  int writeVarInt(int value);

  int readInt();

  void writeLong(int value);

  int readLong();

  void writeFloat(double value);

  double readFloat();

  void writeDouble(double value);

  double readDouble();

  void writeString(String value);

  String readString();

  void writePacket(Object? packet, int protocolId);

  Object readPacket(int protocolId);

  void writeBoolArray(List<bool> array);

  List<bool> readBoolArray();

  void writeByteArray(List<int> array);

  List<int> readByteArray();

  void writeShortArray(List<int> array);

  List<int> readShortArray();

  void writeIntArray(List<int> array);

  List<int> readIntArray();

  void writeLongArray(List<int> array);

  List<int> readLongArray();

  void writeFloatArray(List<double> array);

  List<double> readFloatArray();

  void writeDoubleArray(List<double> array);

  List<double> readDoubleArray();

  void writeStringArray(List<String> array);

  List<String> readStringArray();

  void writePacketArray(List<Object> array, int protocolId);

  List<T> readPacketArray<T>(int protocolId);

  void writeBoolList(List<bool> list);

  List<bool> readBoolList();

  void writeByteList(List<int> list);

  List<int> readByteList();

  void writeShortList(List<int> list);

  List<int> readShortList();

  void writeIntList(List<int> list);

  List<int> readIntList();

  void writeLongList(List<int> list);

  List<int> readLongList();

  void writeFloatList(List<double> list);

  List<double> readFloatList();

  void writeDoubleList(List<double> list);

  List<double> readDoubleList();

  void writeStringList(List<String> list);

  List<String> readStringList();

  void writePacketList(List<Object> list, int protocolId);

  List<T> readPacketList<T>(int protocolId);

  void writeBoolSet(Set<bool> set);

  Set<bool> readBoolSet();

  void writeByteSet(Set<int> set);

  Set<int> readByteSet();

  void writeShortSet(Set<int> set);

  Set<int> readShortSet();

  void writeIntSet(Set<int> set);

  Set<int> readIntSet();

  void writeLongSet(Set<int> set);

  Set<int> readLongSet();

  void writeFloatSet(Set<double> set);

  Set<double> readFloatSet();

  void writeDoubleSet(Set<double> set);

  Set<double> readDoubleSet();

  void writeStringSet(Set<String> set);

  Set<String> readStringSet();

  void writePacketSet(Set<Object>set, int protocolId);

  Set<T> readPacketSet<T>(int protocolId);

  void writeIntIntMap(Map<int, int> map);

  Map<int, int> readIntIntMap();

  void writeIntLongMap(Map<int, int> map);

  Map<int, int> readIntLongMap();

  void writeIntStringMap(Map<int, String> map);

  Map<int, String> readIntStringMap();

  void writeIntPacketMap(Map<int, Object> map, int protocolId);

  Map<int, T> readIntPacketMap<T>(int protocolId);

  void writeLongIntMap(Map<int, int> map);

  Map<int, int> readLongIntMap();

  void writeLongLongMap(Map<int, int> map);

  Map<int, int> readLongLongMap();

  void writeLongStringMap(Map<int, String> map);

  Map<int, String> readLongStringMap();

  void writeLongPacketMap(Map<int, Object> map, int protocolId);

  Map<int, T> readLongPacketMap<T>(int protocolId);

  void writeStringIntMap(Map<String, int> map);

  Map<String, int> readStringIntMap();

  void writeStringLongMap(Map<String, int> map);

  Map<String, int> readStringLongMap();

  void writeStringStringMap(Map<String, String> map);

  Map<String, String> readStringStringMap();

  void writeStringPacketMap(Map<String, Object> map, int protocolId);

  Map<String, T> readStringPacketMap<T>(int protocolId);
}
